extensions/opc/src/putopc.cpp

```diff
@@ -301,8 +301,9 @@ namespace processors {
             break;
           }
           case opc::OPCNodeDataType::Boolean: {
-            bool value;
-            if (utils::StringUtils::StringToBool(contentstr, value)) {
+            utils::optional<bool> ini_value = utils::StringUtils::toBool(contentstr);
+            if (ini_value) {
+              value = ini_value.value();
```

Comment:
This variable was previously shadowed, and now is a string assigned to a boolean value. This is a very big mistake.

Reply:
If you update this line, it will also no longer be a mistake:
```c++
sc = connection_->update_node(targetnode, the_new_name_of_your_optional.value());
```